### ISSUE TITLE
Question about the  implementation of contrastive attention

### ISSUE THREAD

**Issue by OliverHxh** [guxinqian/AP3D#9]
Thanks for your great  work! 
I am studying your code,  and I find that in the implementation of contrastive attention, you use a **detach()** trick like:
```
x_att = self.x_mapping(x.unsqueeze(3).expand(-1, -1, -1, N-1, -1, -1).contiguous().view(b, c, (N-1)*t, h, w).detach())
n_att = self.n_mapping(neighbor_new.detach())
contrastive_att = self.contrastive_att_net(x_att * n_att)
neighbor_new = neighbor_new * contrastive_att
```
which indicates that you don't want the gradients to be broadcast. I want to know the exact reason of the **detach()** usage. And have you tried to train and test your network without **detach()**, how about the performance under such condition? Thank you very much!

**Comment by guxinqian** [comment]
Thanks for your attention to this work.
In fact, there isn't an exact reason, and just using detach() can get a higher testing performance.

**Comment by OliverHxh** [comment]
That makes sense, thank you very much!
